util/types/datum.go

```
@@ -741,14 +727,13 @@ func (d *Datum) convertToFloat(sc *variable.StatementContext, target *FieldType)
 		f, _ = d.GetMysqlDuration().ToNumber().ToFloat64()
 	case KindMysqlDecimal:
 		f, _ = d.GetMysqlDecimal().ToFloat64()
-	case KindMysqlHex:
-		f = d.GetMysqlHex().ToNumber()
-	case KindMysqlBit:
-		f = d.GetMysqlBit().ToNumber()
 	case KindMysqlSet:
 		f = d.GetMysqlSet().ToNumber()
 	case KindMysqlEnum:
 		f = d.GetMysqlEnum().ToNumber()
+	case KindBinaryLiteral, KindMysqlBit:
+		val, _ := d.GetBinaryLiteral().ToInt()
```

error is not handled.